src/Unity.Mathematics/quaternion.cs

```diff
@@ -666,13 +666,7 @@ public static float3 rotate(quaternion q, float3 v)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static quaternion nlerp(quaternion q1, quaternion q2, float t)
         {
-            float dt = dot(q1, q2);
-            if(dt < 0.0f)
-            {
-                q2.value = -q2.value;
-            }
-
-            return normalize(quaternion(lerp(q1.value, q2.value, t)));
+            return normalize(q1.value + t * (chgsign(q2.value, dot(q1, q2)) - q1.value));
```

Comment:
Is there a meaningful difference between this version and the old version of `nlerp`?

Reply:
It will compute the same result however the thing we want to get rid of is the jump instruction:
![image](https://user-images.githubusercontent.com/1699845/141127315-693741c6-534e-4e0f-aa0d-40c0d1557163.png)

Reply:
Oh gotcha, I missed that from your original PR description.